osu.Game/Graphics/UserInterface/RollingCounter.cs

```diff
@@ -46,57 +47,71 @@ public abstract class RollingCounter<T> : Container, IHasAccentColour
         public virtual T DisplayedCount
         {
             get => displayedCount;
-
             set
             {
                 if (EqualityComparer<T>.Default.Equals(displayedCount, value))
                     return;
 
                 displayedCount = value;
-                DisplayedCountSpriteText.Text = FormatCount(value);
+                if (displayedCountSpriteText != null)
+                    displayedCountSpriteText.Text = FormatCount(value);
             }
         }
 
         public abstract void Increment(T amount);
 
+        private float textSize = 40f;
+
         public float TextSize
         {
-            get => DisplayedCountSpriteText.Font.Size;
-            set => DisplayedCountSpriteText.Font = DisplayedCountSpriteText.Font.With(size: value);
+            get => displayedCountSpriteText?.Font.Size ?? textSize;
```

Comment:
why not just remove all this complexity and leave it to the overriding of `CreateSpriteText` to adjust it? is this used at runtime anywhere?

Reply:
`HUDOverlay` uses that on its counter components (combo, score, etc..), though even after removing that there's also `AccentColour` which still needs to exist.

Reply:
i can't read your response

Reply:
So, there are `TextSize` usages in `HUDOverlay`:
https://github.com/ppy/osu/blob/96c4a30facc6bd2b0cdc8415852624e4a8e2c835/osu.Game/Screens/Play/HUDOverlay.cs#L235
https://github.com/ppy/osu/blob/96c4a30facc6bd2b0cdc8415852624e4a8e2c835/osu.Game/Screens/Play/HUDOverlay.cs#L244
https://github.com/ppy/osu/blob/96c4a30facc6bd2b0cdc8415852624e4a8e2c835/osu.Game/Screens/Play/HUDOverlay.cs#L251

they're just initial text sizes so I can move them to inside their `CreateSpriteText()` method.

As for `AccentColour`, I guess it can be removed for now as it's not used anywhere and likely won't be.

Reply:
Sounds good.